pulser-core/pulser/sequence/sequence.py

```diff
@@ -460,6 +465,13 @@ def get_addressed_bases(self) -> tuple[str, ...]:
         """Returns the bases addressed by the declared channels."""
         return tuple(self._basis_ref)
 
+    def get_addressed_states(self) -> list[States]:
+        """Returns the states addressed by the declared channels."""
+        all_states = set().union(
+            *(set(EIGENSTATES[basis]) for basis in self.get_addressed_bases())
+        )
+        return [state for state in STATES_RANK if state in all_states]
```

Comment:
I feel like there is a bit of code replication in this block, with small variations. Do you think we could make a single function that takes in a list of bases and returns the ordered states? If it makes sense, we could even make it a static method

Reply:
I have added a function to tackle this, let me know what you think (and what you think of the name of this function as well ^^')